### ISSUE TITLE
Failed update check

### ISSUE THREAD

**Issue by ahernandezmiro** [trembon/switch-library-manager#63]
I was using original 1.4 version until for some reason it stopped showing game updates (tab shows as all games updated & no new updates available, even though I know there are missing updates).
I have upgraded to your 1.8 version and the problem persists. Basically all games appear as updated even when it's not the case.

titles.json & versions.json files are good.

if there is any other information I can provide please let me know

**Comment by trembon** [comment]
is it the same result when doing a hard rescan?

also with the versions are good, have you verified that the games you have do have an update entry in the versions.json file? you can search in the file with the game id to find its entry

**Comment by ahernandezmiro** [comment]
I have performed a hard rescan and the problem persists. The titles are in the versions file and show updates available. I am attaching a screenshot where you can see the issue.
I removed the last update for BotW from my local files, did a hard rescan, and the missing updates tab still appears empty.
I am attaching a couple screenshots where you can see the problem:

![Sin título](https://github.com/user-attachments/assets/a422d5e9-afdb-4c97-8466-fd67dda3bef9)
![Sin título2](https://github.com/user-attachments/assets/0aa385de-cf4e-463a-b7b8-f2cfcfe72ef5)

**Comment by trembon** [comment]
thanks for the information, i will see if i can find anything

**Comment by ahernandezmiro** [comment]
If you could share some steps to set up the development environment for the project I can try to help. Neither JS nor Golang are my speciality but surely I can do something

**Comment by larg000** [comment]
it looks like a bug i had  and already post here

check in settings.json  if "titles_json_url" and "versions_json_url" are not empty

value should be this :
"versions_json_url": "https://raw.githubusercontent.com/blawar/titledb/master/versions.json",
 "titles_json_url": "https://tinfoil.media/repo/db/titles.json",

for some reason this value are erased and become empty
i dont know why

use notepad to make the change

**Comment by ahernandezmiro** [comment]
My settings.json already has both urls set to the values you shared. Attaching file for reference.
[settings.json](https://github.com/user-attachments/files/16985019/settings.json)

**Comment by larg000** [comment]
did you try a fresh install ?

**Comment by ahernandezmiro** [comment]
yes, I've tested removing titles, versions and settings.json. had to add titles manually as tinfoil url is down.

**Comment by ahernandezmiro** [comment]
nothing?

**Comment by trembon** [comment]
i havent been able to reproduce this error and other people dont seem to have this problem either.
so currently i'm not sure what i'm able to do about this, unfortunately

**Comment by ahernandezmiro** [comment]
I think the application has problems when downloading titles.json & reading a huge amount of files from an external HDD (>500 games, >4TB). I have tried the newer 1.9 and the app closed itself while analyzing files at around 33%. It was also unable to fetch titles from the default url, don't know if it's just my country but that domain has been shut down for long. When I manually added titles.json and tried to relaunch the exe, it would just close automatically.
